pkg/template/resource.go

```diff
@@ -83,5 +88,16 @@ func ApplyParamsToResourceTemplate(params []pipelinev1.Param, rt json.RawMessage
 func applyParamToResourceTemplate(param pipelinev1.Param, rt json.RawMessage) json.RawMessage {
 	// Assume the param is valid
 	paramVariable := fmt.Sprintf("$(params.%s)", param.Name)
-	return []byte(strings.Replace(string(rt), paramVariable, param.Value.StringVal, -1))
+	return bytes.Replace(rt, []byte(paramVariable), []byte(param.Value.StringVal), -1)
+}
+
+// Uid generates a postfix like the k8s apiserver does for the generateName field.
```

Comment:
nit: this function generates a random string, not a postfix; the random string can be used as a postfix, but that's up to the user's discretion

maybe should be something like: `Uid generates a random string, like the postfix generated by the Kubernetes generateName metadata field.`